src/main/java/me/pugabyte/nexus/features/minigames/models/mechanics/multiplayer/teams/TeamMechanic.java

```diff
@@ -201,4 +207,214 @@ public void onQuit(MatchQuitEvent event) {
 		super.onQuit(event);
 	}
 
+	public boolean basicBalanceCheck(List<Minigamer> minigamers) {
+		if (minigamers.isEmpty())
+			return false;
+
+		Match match = minigamers.get(0).getMatch();
+		Arena arena = match.getArena();
+		List<Team> teams = new ArrayList<>(arena.getTeams());
+
+		int required = 0;
+		for (Team team : teams) required += team.getMinPlayers();
+
+		if (match.getMinigamers().size() < required) {
+			criticalErrorAbort("Not enough players to meet team requirements!", match);
+			return false;
+		}
+
+		return true;
+	}
+
+	protected List<BalanceWrapper> getBalanceWrappers(Minigamer minigamer) {
+		return getBalanceWrappers(minigamer.getMatch());
+	}
+
+	protected List<BalanceWrapper> getBalanceWrappers(Match match) {
+		// ALL PERCENTAGES HERE RANGE FROM 0 to 1 !!
+		List<Team> teams = match.getArena().getTeams();
+		List<BalanceWrapper> wrappers = new ArrayList<>();
+		double percentageSum = 0; // sum of all balance percentages
+		int noPercentage = 0; // count of teams w/o balance percentages
+		for (Team team : teams) {
+			BalanceWrapper wrapper = new BalanceWrapper(team, match);
+			wrappers.add(wrapper);
+			if (wrapper.getPercentage() != null)
+				percentageSum += wrapper.getPercentage();
+			else
+				noPercentage++;
+		}
+
+		if (noPercentage > 0 && percentageSum < 1) {
+			// evenly split the balance of teams that don't have a balance percentage (if there is any unassigned %)
+			double percentage = (1d/noPercentage) * (1d-percentageSum);
+			wrappers.stream().filter(wrapper -> wrapper.getPercentage() == null).forEach(wrapper -> wrapper.setPercentage(percentage));
+		}
+
+		// ensure percentages add up to 100
+		double totalPercentage = wrappers.stream().mapToDouble(BalanceWrapper::getPercentage).sum();
+		wrappers.forEach(wrapper -> wrapper.setPercentage(wrapper.getPercentage() / totalPercentage));
+
+		return wrappers;
+	}
+
+	@Override
+	public void onDeath(MinigamerDeathEvent event) {
+		// auto-balancing
+		Match match = event.getMatch();
+		Minigamer minigamer = event.getMinigamer();
+		super.onDeath(event);
+		if (!minigamer.isAlive() || match.isEnded())
+			return;
+
+		List<BalanceWrapper> wrappers = getBalanceWrappers(match).stream().filter(w -> !w.getTeam().equals(minigamer.getTeam()) && w.percentageDiscrepancy() > 0 && w.extraPlayerPercentDiscrepancy() >= 0).collect(Collectors.toList());
+		if (wrappers.isEmpty())
+			return;
+		// sort teams by closest to being equal
+		wrappers.sort((w1, t1) -> (int) ((w1.extraPlayerPercentDiscrepancy() - t1.extraPlayerPercentDiscrepancy())*100));
```

Comment:
Is this sorting not already implemented in the comparable override?

Reply:
the comparable sorts by `.percentageDiscrepancy()` and in the other direction, but you're right, you'd get the same results by using the comparable and reversing the list